### PR TITLE
Cosmetic changes (code style, documentation, etc.)

### PR DESCRIPTION
**Pull request by saharNooby**
Most changes are cosmetic and do not affect APIs and performance.

`ggml` was updated to the latest commit. I'm pretty sure there were no breaking changes. I don't expect any impact on performance.

### CODE REVIEW

File: tests/test_context_cloning.c [hunk 1]
```diff
@@ -42,12 +42,12 @@ int main() {
 
 	rwkv_eval(ctx, prompt[0], NULL, state, logits);
 
-	for (const unsigned char * token = prompt + 1; *token != 0; token++) {
-		rwkv_eval(ctx, *token, state, state, logits);
+	for (int i = 1; i < prompt[i] != 0; i++) {
```
Comment (LoganDark) [inline tests/test_context_cloning.c:45]: `i < prompt[i] != 0` huh?

File: tests/test_context_cloning.c [hunk 2]
```diff
@@ -21,16 +21,16 @@ int main() {
...
 
-	// 0xd1 or 209 is space (0x20 or \u0120 in tokenizer)
-	const unsigned char * prompt = "hello\xd1world";
+	const unsigned char prompt[12] = "hello world";
 
 	rwkv_eval(ctx, prompt[0], NULL, state, logits);
 
-	for (const unsigned char * token = prompt + 1; *token != 0; token++) {
-		rwkv_eval(ctx, *token, state, state, logits);
+	for (int i = 1; prompt[i] != 0; i++) {
```
Comment (LoganDark) [inline tests/test_context_cloning.c:28]: should this be just `prompt[i]`, not `prompt[i] != 0`?

File: tests/test_tiny_rwkv.c [hunk 3]
```diff
@@ -33,7 +33,7 @@ void test_model(const char * model_path, const float * expected_logits, const fl
     ASSERT(rwkv_gpu_offload_layers(model, N_GPU_LAYERS), "Unexpected error %d", rwkv_get_last_error(model));
 #endif
 
-    uint32_t n_vocab = rwkv_get_logits_len(model);
+    size_t n_vocab = rwkv_get_logits_len(model);
```
Comment (LoganDark) [inline tests/test_tiny_rwkv.c:36]: `const size_t` I suppose

File: tests/test_context_cloning.c (hunk 1 above)
Comment (saharNooby) [reply tests/test_context_cloning.c:45]: LOL

File: tests/test_context_cloning.c (hunk 2 above)
Comment (saharNooby) [reply tests/test_context_cloning.c:28]: I think this would be a valid simplification, but explicit comparison feels more clear to me.

Comment (LoganDark) [reply tests/test_context_cloning.c:28]: It's a common practice in C when reading strings, for what it's worth, but either way is fine I think

File: tests/test_tiny_rwkv.c (hunk 3 above)
Comment (saharNooby) [reply tests/test_tiny_rwkv.c:36]: I'll add it, but what will this accomplish? :)

Comment (LoganDark) [reply tests/test_tiny_rwkv.c:36]: It makes the variable immutable

**Review by LoganDark — APPROVED** [review]

**Pull request merged**